### PR TITLE
MJAVADOC-431 allow maven descriptor to javadoc jar

### PR DESCRIPTION
**Pull request by peterlynch**
Note: ~~maven-archiver dependency version was upgraded to take advantage of MavenProject.clone(), which avoids having the new unit test from creating a maven-archiver directory in the basedir because the MavenProject() constructor used in 2.5 does not copy the build element.~~ reverted

### CODE REVIEW

**Comment by michael-o** [comment]
Several issues with the PR commented inline.

File: maven-javadoc-plugin/pom.xml [hunk 1]
```diff
@@ -148,7 +148,7 @@ under the License.
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-archiver</artifactId>
-      <version>2.5</version>
+      <version>2.6</version>
```
Comment (michael-o) [inline maven-javadoc-plugin/pom.xml]: That should be a separate ticket and not just a mere change in this PR.

File: maven-javadoc-plugin/src/main/java/org/apache/maven/plugin/javadoc/JavadocArchiveConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * @see <a href="https://issues.apache.org/browse/MJAVADOC-431">MJAVADOC-431</a>
+ */
+public class JavadocArchiveConfiguration
+        extends MavenArchiveConfiguration
+{
+    public JavadocArchiveConfiguration()
+    {
+        setAddMavenDescriptor( false );
+    }
+}
```
Comment (michael-o) [inline maven-javadoc-plugin/src/main/java/org/apache/maven/plugin/javadoc/JavadocArchiveConfiguration.java]: Add an empty trailing line.

File: maven-javadoc-plugin/src/test/java/org/apache/maven/plugin/javadoc/JavadocJarTest.java [hunk 3]
```diff
@@ -109,29 +103,62 @@ else if ( Float.compare( javadocVersion, 1.8f ) < 0 )
...
+        expected.add("META-INF/maven/org.apache.maven.plugins.maven-javadoc-plugin.unit/javadocjar-archive-config/");
+        expected.add("META-INF/maven/org.apache.maven.plugins.maven-javadoc-plugin.unit/javadocjar-archive-config/pom.xml");
+        expected.add("META-INF/maven/org.apache.maven.plugins.maven-javadoc-plugin.unit/javadocjar-archive-config/pom.properties");
+
+        for (int i = 0; i < expected.size(); i++) {
+            String entry = expected.get(i);
+            assertTrue("Expected jar to contain " + entry, set.contains(entry));
+        }
     }
 }
```
Comment (michael-o) [inline maven-javadoc-plugin/src/test/java/org/apache/maven/plugin/javadoc/JavadocJarTest.java:164]: The entire file has been unproperly reformatted. Please restore the code style enforced by Apache Maven as well as your new tests.

File: maven-javadoc-plugin/src/test/java/org/apache/maven/plugin/javadoc/stubs/JavadocJarArchiveConfigProjectStub.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    extends MavenProjectStub
+{
+    private Scm scm;
+
+    public JavadocJarArchiveConfigProjectStub()
+    {
+        File projectFile =  new File( getBasedir(), "javadocjar-archive-config.xml" );
+        readModel( new File( getBasedir(), "javadocjar-archive-config.xml" ) );
+
+        setFile(projectFile);
```
Comment (michael-o) [inline maven-javadoc-plugin/src/test/java/org/apache/maven/plugin/javadoc/stubs/JavadocJarArchiveConfigProjectStub.java]: Format.

File: maven-javadoc-plugin/src/test/java/org/apache/maven/plugin/javadoc/stubs/JavadocJarDefaultMavenProjectStub.java [hunk 5]
```diff
@@ -59,7 +59,7 @@ public JavadocJarDefaultMavenProjectStub()
         setArtifact( artifact );
 
         Build build = new Build();
-        build.setFinalName( "javadocjar-default" );
+        build.setFinalName( "javadocjar-archive-config" );
```
Comment (michael-o) [inline maven-javadoc-plugin/src/test/java/org/apache/maven/plugin/javadoc/stubs/JavadocJarDefaultMavenProjectStub.java]: If this is default why has this been changed?

File: maven-javadoc-plugin/src/test/resources/unit/javadocjar-archive-config/javadocjar/def/App.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+     * Sample method
+     *
+     * @param str   the string to be displayed
+     */
+    protected void sampleMethod( String str )
+    {
+        System.out.println( str );
+    }
+
+}
```
Comment (michael-o) [inline maven-javadoc-plugin/src/test/resources/unit/javadocjar-archive-config/javadocjar/def/App.java]: Trailing line.

**Comment by michael-o** [comment]
@peterlynch Please update your PR according to me comments and I will review again.

File: maven-javadoc-plugin/pom.xml (hunk 1 above)
Comment (peterlynch) [inline maven-javadoc-plugin/pom.xml]: Do you want a separate jira issue but include this change only in this PR?

Or do you want a new PR/issue with this change only?

Or do you want me to revert this version change ( which will cause a **maven-archiver** directory to be created in the root of the project after the tests run ).

File: maven-javadoc-plugin/src/test/java/org/apache/maven/plugin/javadoc/stubs/JavadocJarDefaultMavenProjectStub.java (hunk 5 above)
Comment (peterlynch) [inline maven-javadoc-plugin/src/test/java/org/apache/maven/plugin/javadoc/stubs/JavadocJarDefaultMavenProjectStub.java]: oops

**Comment by peterlynch** [comment]
@michael-o how does it look now?

File: maven-javadoc-plugin/pom.xml (hunk 1 above)
Comment (michael-o) [inline maven-javadoc-plugin/pom.xml]: Drop this change. If you want to upate maven archiver, create a separate JIRA ticket for that.

**Comment by michael-o** [comment]
@peterlynch Please replay your changes on top of master again instead of pushing changes/commits. Squash into one and I will review and merge.

**Comment by michael-o** [comment]
@peterlynch I am still waiting for an updated PR.

**Comment by peterlynch** [comment]
@michael-o replaced by #70

**Pull request closed without merging**